packages/core/src/dateParser.ts

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Cognite AS
+
+import { cloneDeepWith } from 'lodash';
+import { CDFHttpClient } from './httpClient/cdfHttpClient';
+import { HttpRequestOptions, HttpResponse } from './httpClient/basicHttpClient';
+
+export class DateParser {
+  /**
+   * Provides functions for turning dates into numbers and
+   * vice versa in objects.
+   * Will indiscriminatly convert Date objects to numbers.
+   * Will only convert from number/string to Date if the path
+   * looks like:
+   *
+   *  lead.lead.lead.date
+   *
+   * where lead can be any field from the lead parameter,
+   * and the date is a field from the dates parameter.
+   * Lead can also be the name of an array, which will allow all elements of that array
+   *
+   *  lead[5].lead.date
+   *
+   * @param lead the keys that lead to dates
+   * @param dates the keys that can be parsed as Date objects
+   */
+  constructor(private lead: string[], private dates: string[]) {}
```

Comment:
```suggestion
  constructor(private path: string[], private datePropNames: string[]) {}
```
and now you don't even need to document it anymore